pkg/config/state/storage/yaml/yaml_storage.go

```diff
@@ -0,0 +1,68 @@
+package yaml
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/ddev/ddev/pkg/config/state"
+	"github.com/ddev/ddev/pkg/config/state/types"
+	"gopkg.in/yaml.v3"
+)
+
+// New returns a new StateStorage interface based on a YAML file.
+//
+// Parameter 'stateFilePath' is the path and file name to the storage file.
+func New(stateFilePath string) types.StateStorage {
```

Comment:
Wow could this be nice. So many places.

Reply:
> https://github.com/mitchellh/mapstructure looks like a wonderful thing. Could be used several other places.

Yes, makes some things easier and more flexible compared to the json and yaml implementation.

Reply:
He he, factory functions :) Easier to use with strict typed parameters compared to a manual setup of the struct. And you also can call methods e.g. or do other important init stuff.